donate/payments/views.py

```diff
@@ -309,7 +309,9 @@ def process_monthly_transaction(self, form):
             'merchant_account_id': get_merchant_account_id_for_card(self.currency),
             'payment_method_token': payment_method.token,
             'price': form.cleaned_data['amount'],
-            'first_billing_date': now().date(),
+            'options': {
```

Comment:
Based on this message from Braintree support:

"I’m Shaquille with Braintree Technical Support, happy to help.
I noted in the subscription examples provided the subscription was created with a specified start date in the future. This creates a subscription that remains Pending until the first billing date when the customer is charged and the subscription becomes Active. Do note, you can [create a subscription to start billing immediately](https://developer.paypal.com/braintree/docs/reference/request/subscription/create#set-start-date). This revision can be made to the plan that subscriptions are based on, or in the subscription.create() call itself."

It sounds like we are given two options when creating a new subscription: 
- Set the date and time for the subscription ourselves, which is what we were originally doing
- Or set the subscription to start immediately, which is what this PR is introducing. 

The reason for the change being that if a user submits a recurring donation past a certain time (around 7:30PM EST according to Jackie Lu) the user experiences a delay in recieving a donation receipt. 

Will Easton has also confirmed that when the user makes a donation at this time, the donation is set to "Pending" in braintree until the following day.

We believe this is due to differences in time zone between our application and Braintree. 

I believe that setting the donation to begin immediately will resolve this problem as it removes timezones from the equation.

Reply:
Great description, thanks!

Reply:
@danielfmiranda please make sure that this change applies ONLY to recurring donations that are made directly from the donate page, and NOT those that are done via the pop-up modal that asks for a recurring donation (which has to start in one month) after you give a one-time gift of (I think) $15 or more.

Reply:
(@danielfmiranda if this doesn't make sense, please ping me on slack.)

Reply:
Hi @WillatMozFdn! 

I can confirm that this change does not make any changes to the donations made in the pop-up modal that appears after making a one time donation. 

Here is a link to the ["Upsell-View" ](https://github.com/mozilla/donate-wagtail/blob/29a0ec9fdb4e45d4204d5d38175044c1897de961/donate/payments/views.py#L596)which is displayed after a user has made a one time donation. we are still setting the next recurring donation to one month from the day the user made the initial one-time donation, should they choose to create a recurring one. 

Please let me know if you have any questions, thank you!

Reply:
Perfect, Daniel, thank you!  I had just wanted to triple-check on that ... because if we set ALL the recurrings to start immediately (both modal and regular), we'd have a whole separate set of donor care problems on our hands.  :-)  Thanks so much!!